Rebus.RabbitMq/RabbitMq/ConnectionManager.cs

```diff
@@ -259,10 +259,18 @@ static SslOption ToSslOption(SslSettings ssl)
                 properties["User"] = userDomainName;
             }
 
-            var currentProcess = Process.GetCurrentProcess();
+            //TODO: Ideally this should be ported to retrieve AppDomain.CurrentDomain info for netframework and System.Reflection.Assembly information for netstandard
+            try
+            {
+                var currentProcess = Process.GetCurrentProcess(); //This may throw a SecurityException if running from an IIS app. pool process
 
-            properties.Add("ProcessName", currentProcess.ProcessName);
-            properties.Add("FileName", currentProcess.MainModule.FileName);
+                properties.Add("ProcessName", currentProcess.ProcessName);
+                properties.Add("FileName", currentProcess.MainModule.FileName);
+            }
+            catch (System.Security.SecurityException)
+            {
+                // Insufficient permissions to get process information
```

Comment:
wouldn't it be neat to do something like this:

    properties.Add("ProcessName", "Could not be determined.");
    properties.Add("FileName", "Could not be determined.");

?

Reply:
Sure that's even better as a quick fix for now. Although I think the ideal solution is to retrieve that info in the way described in the comments but that will need further testing. Thanks for the quick reply!